scripts/wait-on-circle-jobs.js

```diff
@@ -0,0 +1,226 @@
+  //   id: '566ffe9a-62d4-45cd-9a27-9882139e0121',
+  //   name: 'linux',
+  //   project_slug: 'gh/cypress-io/cypress',
+  //   status: 'failed',
+  //   started_by: '45ae8c6a-4686-4e71-a078-fb7a3b9d9e59',
+  //   pipeline_number: 12461,
+  //   created_at: '2020-07-20T19:45:41Z',
+  //   stopped_at: '2020-07-20T20:06:54Z'
+  // }
+
+  return response
+}
+
+/**
+ * Job status
+ *  - blocked (has not run yet)
+ *  - running (currently running)
+ *  - failed | success
+*/
+const getJobStatus = async (workflowId) => {
+  const auth = getAuth()
+  // typo at https://circleci.com/docs/2.0/api-intro/
+  // to retrieve all jobs, the url is "/workflow/:id/job"
+  const url = `https://${auth}@circleci.com/api/v2/workflow/${workflowId}/job`
+  const response = await got(url).json()
+
+  // returns something like
+  // {
+  //   next_page_token: null,
+  //   items: [
+  //     {
+  //       dependencies: [],
+  //       job_number: 400959,
+  //       id: '7021bcc7-90c1-47d9-bf99-c0372a4f8f49',
+  //       started_at: '2020-07-20T19:45:46Z',
+  //       name: 'build',
+  //       project_slug: 'gh/cypress-io/cypress',
+  //       status: 'success',
+  //       type: 'build',
+  //       stopped_at: '2020-07-20T19:50:07Z'
+  //     }
+  //   ]
+  // }
+  return response
+}
+
+const waitForAllJobs = async (jobNames, workflowId) => {
+  let response
+
+  try {
+    response = await getJobStatus(workflowId)
+  } catch (e) {
+    console.error(e)
+    process.exit(1)
+  }
+
+  // if a job is pending, its status will be "blocked"
+  const blockedJobs = _.filter(response.items, { status: 'blocked' })
+  const failedJobs = _.filter(response.items, { status: 'failed' })
+
+  const runningJobs = _.filter(response.items, { status: 'running' })
+
+  const blockedJobNames = _.map(blockedJobs, 'name')
+  const runningJobNames = _.map(runningJobs, 'name')
+  const failedJobNames = _.map(failedJobs, 'name')
+
+  if (requireAllJobsToPass && _.intersection(jobNames, failedJobNames).length) {
+    console.error('At least one failing job has prevented percy-finalize from running', failedJobs)
+    process.exit(1)
+  }
+
+  debug('failed jobs %o', _.map(failedJobs, 'name'))
+  debug('blocked jobs %o', blockedJobNames)
+  debug('running jobs %o', runningJobNames)
+
+  if (!runningJobs.length || (runningJobs.length === 1 && runningJobs[0].name === jobName)) {
+    // there are no more jobs to run, or this is the last running job
+    console.log('all jobs are done, finishing this job')
+
+    return Promise.resolve()
+  }
+
+  const futureOrRunning = _.union(blockedJobs, runningJobNames)
+  const jobsToWaitFor = _.intersection(jobNames, futureOrRunning)
+
+  debug('jobs to wait for %o', jobsToWaitFor)
+
+  if (!jobsToWaitFor.length) {
+    console.log('No more jobs to wait for!')
+
+    return Promise.resolve()
+  }
+
+  return Promise.reject(new Error('Jobs have not finished'))
+}
+
+const waitForJobToPass = Promise.method(async (jobName, workflow = workflowId) => {
```

Comment:
I know most of this file was just reinstated, but it doesn't look like this function is used anymore. Can we remove this and anything else that's not actively used?

Reply:
Good point, updated 👍